main_test.go

```diff
@@ -209,6 +209,16 @@ kube_pod_container_status_terminated_reason{namespace="default",pod="pod0",conta
 kube_pod_container_status_terminated_reason{namespace="default",pod="pod0",container="container3",reason="Completed"} 0
 kube_pod_container_status_terminated_reason{namespace="default",pod="pod0",container="container3",reason="Error"} 0
 kube_pod_container_status_terminated_reason{namespace="default",pod="pod0",container="container3",reason="ContainerCannotRun"} 0
+# HELP kube_pod_container_status_terminated_reasons_total Counter with observed container restarts and reason the container was terminated.
+# TYPE kube_pod_container_status_terminated_reasons_total counter
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container2",reason="Completed"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container2",reason="ContainerCannotRun"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container2",reason="Error"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container2",reason="OOMKilled"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container3",reason="Completed"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container3",reason="ContainerCannotRun"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container3",reason="Error"} 0
+kube_pod_container_status_terminated_reasons_total{namespace="default",pod="pod0",container="container3",reason="OOMKilled"} 0
 # HELP kube_pod_container_status_last_terminated_reason Describes the last reason the container was in terminated state.
 # TYPE kube_pod_container_status_last_terminated_reason gauge
 kube_pod_container_status_last_terminated_reason{namespace="default",pod="pod0",container="container2",reason="OOMKilled"} 1
```

Comment:
Once add new object metrics, we also need to complete the [doc](https://github.com/kubernetes/kube-state-metrics/tree/master/Documentation) in the corresponding places. Thanks for your work!

Reply:
thanks for the reminder, I will wait for the discussion about this proposal to continue for a little bit and then based on the output, either add docs if the PR provides additional value or close if it isn't worth having.

Reply:
thanks for the reminder, added to the docs :)